### DISCUSSION TITLE
[OP Stack Deployment] Getting an error when deploying the OP Stack Contracts

### DISCUSSION THREAD

**Discussion by yohanelly95** [ethereum-optimism/developers#308] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

My goal is to deploy an OP Stack Chain using Celestia as DA layer. I am running a celestia light node on the testnet mocha, but the contracts deployments seem to be failing. Ive tried with the default chainID 42069 and it still fails https://docs.optimism.io/builders/chain-operators/tutorials/create-l2-rollup#deploy-the-l1-contracts. Running `direnv allow` does not fix this (does not seem salt related). 

### Logs

```│   └─ ← ()
    ├─ [0] VM::label(DeployConfig: [0x9568d36E291c2C4c34fa5593fcE73715abEf6F9c], "DeployConfig")
    │   └─ ← ()
    ├─ [0] VM::allowCheatcodes(DeployConfig: [0x9568d36E291c2C4c34fa5593fcE73715abEf6F9c])
    │   └─ ← ()
    ├─ [2700822] DeployConfig::read("/root/optimism/packages/contracts-bedrock/deploy-config/getting-started.json")
    │   ├─ [0] console::log("DeployConfig: reading file %s", "/root/optimism/packages/contracts-bedrock/deploy-config/getting-started.json") [staticcall]
    │   │   └─ ← ()
    │   ├─ [0] VM::readFile("/root/optimism/packages/contracts-bedrock/deploy-config/getting-started.json") [staticcall]
    │   │   └─ ← <file>
    │   ├─ [0] VM::parseJsonAddress("<stringified JSON>", "$.finalSystemOwner") [staticcall]
    │   │   └─ ← 0x150c45CDd6b2ad9b35EFde4D7C2c787C0c7FC53d
    │   ├─ [0] VM::parseJsonAddress("<stringified JSON>", "$.superchainConfigGuardian") [staticcall]
    │   │   └─ ← 0x150c45CDd6b2ad9b35EFde4D7C2c787C0c7FC53d
    │   ├─ [0] VM::parseJsonUint("<stringified JSON>", "$.l1ChainID") [staticcall]
    │   │   └─ ← 11155111 [1.115e7]
    │   ├─ [0] VM::parseJsonUint("<stringified JSON>", "$.l2ChainID") [staticcall]
    │   │   └─ ← 42069 [4.206e4]
    │   ├─ [0] VM::parseJsonUint("<stringified JSON>", "$.l2BlockTime") [staticcall]
    │   │   └─ ← 2
    │   ├─ [0] VM::parseJsonUint("<stringified JSON>", "$.maxSequencerDrift") [staticcall]
    │   │   └─ ← 600
    │   ├─ [0] VM::parseJsonUint("<stringified JSON>", "$.sequencerWindowSize") [staticcall]
    │   │   └─ ← 3600
    │   ├─ [0] VM::parseJsonUint("<stringified JSON>", "$.channelTimeout") [staticcall]
    │   │   └─ ← 300
    │   ├─ [0] VM::parseJsonAddress("<stringified JSON>", "$.p2pSequencerAddress") [staticcall]
    │   │   └─ ← 0x267D36C93e246EAFea2832ca21D94294f0F1ca62
    │   ├─ [0] VM::parseJsonAddress("<stringified JSON>", "$.batchInboxAddress") [staticcall]
    │   │   └─ ← 0xFf00000000000000000000000000000000042069
    │   ├─ [0] VM::parseJsonAddress("<stringified JSON>", "$.batchSenderAddress") [staticcall]
    │   │   └─ ← 0x3876B82bb1eFdFE4130F1334b2B78989e119188C
    │   ├─ [0] VM::parseJsonUint("<stringified JSON>", "$.l2OutputOracleSubmissionInterval") [staticcall]
    │   │   └─ ← 120
    │   ├─ [0] VM::parseJsonInt("<stringified JSON>", "$.l2OutputOracleStartingTimestamp") [staticcall]
    │   │   └─ ← 1711025376 [1.711e9]
    │   ├─ [0] VM::parseJsonUint("<stringified JSON>", "$.l2OutputOracleStartingBlockNumber") [staticcall]
    │   │   └─ ← 0
    │   ├─ [0] VM::parseJsonAddress("<stringified JSON>", "$.l2OutputOracleProposer") [staticcall]
    │   │   └─ ← 0x33d20989622C67a7ddd415F5039cedF8c2dB0fb6
    │   ├─ [0] VM::parseJsonAddress("<stringified JSON>", "$.l2OutputOracleChallenger") [staticcall]
    │   │   └─ ← 0x150c45CDd6b2ad9b35EFde4D7C2c787C0c7FC53d
    │   ├─ [0] VM::parseJsonUint("<stringified JSON>", "$.finalizationPeriodSeconds") [staticcall]
    │   │   └─ ← 12
    │   ├─ [0] VM::parseJsonBool("<stringified JSON>", "$.fundDevAccounts") [staticcall]
    │   │   └─ ← no matching value found at "$.fundDevAccounts"
    │   └─ ← no matching value found at "$.fundDevAccounts"
    └─ ← no matching value found at "$.fundDevAccounts"



== Logs ==
  Using deploy artifact /root/optimism/packages/contracts-bedrock/deployments/getting-started/.deploy
  Connected to network with chainid 11155111
  DeployConfig: reading file /root/optimism/packages/contracts-bedrock/deploy-config/getting-started.json
Error:
script failed: <empty revert data>```

### Additional Information

Is this specific to the celestia/optimism repo? Or is there something I need to fix ? 


To run a custom testnet chain that uses OP Stack Chain and Celestia as the DA I can change the getting-started.json chainID to a new one. Is there anything else needed?

### Feedback

_No response_

**Comment by yohanelly95** [comment]
I see the issue is related to funding devnet accounts, I would like to instead run a testnet OP Stack chain with celestia as DA instead. How do I go about doing this? I have checked celestia docs, ran everything until they redirect to OP docs to run a node.

**Reply by 1195597989** [reply]
https://docs.celestia.org/developers/optimism-devnet

After running according to the document, are there any errors?

**Reply by yohanelly95** [reply]
No errors here at all @1195597989  the celestia node is running fine and I reached the end of the doc where its redirected to Deploy to L1 or L2 and sends me to OP docs

**Comment by yohanelly95** [comment]
also should the `HOME/optimism/ops-bedrock/docker-compose.yml` da section have `mocha` or `mocha-4` as the network ? @1195597989 
this is my docker-compose-yml
```
da:
    image: ghcr.io/rollkit/celestia-da:v0.12.9
    command: >
      celestia-da light start
      --p2p.network=mocha
      --da.grpc.namespace=000008e5f679bf7116cb
      --da.grpc.listen=0.0.0.0:26650
      --core.ip rpc-mocha.pops.one
      --gateway
    environment:
      - NODE_TYPE=light
      - P2P_NETWORK=mocha
    ports:
      - "26650:26650"
      - "26658:26658"
      - "26659:26659"
    volumes:
      - $HOME/.celestia-light-mocha/:/home/celestia/.celestia-light-mocha/

**Reply by 1195597989** [reply]
The yml looks fine

**Reply by yohanelly95** [reply]
There are 3 new variables in the `DeployConfig.s.sol` 
"fundDevAccounts", "useFaultProofs", "expectedGameType" which are not present in the `celestia-develop` branch it seems. So the OP Contract deployments are not working. @1195597989

**Comment by AlfredYyong** [comment]
which github branch for your op-stack?

**Reply by yohanelly95** [reply]
using the `celestia-develop` branch from https://github.com/celestiaorg/optimism. I was guided here from Data Availability Hacks on OP docs.

**Reply by AlfredYyong** [reply]
me too

**Comment by k0de26** [comment]
Your, error message <empty revert data> indicates that your deployment script has failed, but without providing a specific reason why. This makes it a bit trickier to pinpoint the exact problem.

I think your chainid is set incorrectly try this [chainid](https://docs.optimism.io/chain/networks#op-sepolia) instead. 

Also can you provide the code for your `getting-started.json`?

**Reply by yohanelly95** [reply]
> Your, error message indicates that your deployment script has failed, but without providing a specific reason why. This makes it a bit trickier to pinpoint the exact problem.
> 
> I think your chainid is set incorrectly try this [chainid](https://docs.optimism.io/chain/networks#op-sepolia) instead.
> 
> Also can you provide the code for your `getting-started.json`?

@k0de26  So I have used Sepolia as the L1 exectution layer and I am running a custom OP Stack Chain that will run as its own L2, using Celestia mocha testnet as the DA Layer. 

This is my `getting-started.json`: 
```
{
  "l1StartingBlockTag": "0x6c2c4d6f66cb528f7dd77ca8f62de9e64262d22c7623d66e91ebbf0c2954bb9f",

  "l1ChainID": 11155111,
  "l2ChainID": 12202,
  "l2BlockTime": 2,
  "l1BlockTime": 12,

  "maxSequencerDrift": 600,
  "sequencerWindowSize": 3600,
  "channelTimeout": 300,

  "p2pSequencerAddress": "0x267D36C93e246EAFea2832ca21D94294f0F1ca62",
  "batchInboxAddress": "0xff00000000000000000000000000000000042069",
  "batchSenderAddress": "0x3876B82bb1eFdFE4130F1334b2B78989e119188C",

  "l2OutputOracleSubmissionInterval": 120,
  "l2OutputOracleStartingBlockNumber": 0,
  "l2OutputOracleStartingTimestamp": 1711025376,

  "l2OutputOracleProposer": "0x33d20989622C67a7ddd415F5039cedF8c2dB0fb6",
  "l2OutputOracleChallenger": "0x150c45CDd6b2ad9b35EFde4D7C2c787C0c7FC53d",

  "finalizationPeriodSeconds": 12,

  "proxyAdminOwner": "0x150c45CDd6b2ad9b35EFde4D7C2c787C0c7FC53d",
  "baseFeeVaultRecipient": "0x150c45CDd6b2ad9b35EFde4D7C2c787C0c7FC53d",
  "l1FeeVaultRecipient": "0x150c45CDd6b2ad9b35EFde4D7C2c787C0c7FC53d",
  "sequencerFeeVaultRecipient": "0x150c45CDd6b2ad9b35EFde4D7C2c787C0c7FC53d",
  "finalSystemOwner": "0x150c45CDd6b2ad9b35EFde4D7C2c787C0c7FC53d",
  "superchainConfigGuardian": "0x150c45CDd6b2ad9b35EFde4D7C2c787C0c7FC53d",

  "baseFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
  "l1FeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
  "sequencerFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
  "baseFeeVaultWithdrawalNetwork": 0,
  "l1FeeVaultWithdrawalNetwork": 0,
  "sequencerFeeVaultWithdrawalNetwork": 0,

  "gasPriceOracleOverhead": 2100,
  "gasPriceOracleScalar": 1000000,

  "enableGovernance": true,
  "governanceTokenSymbol": "OP",
  "governanceTokenName": "Optimism",
  "governanceTokenOwner": "0x150c45CDd6b2ad9b35EFde4D7C2c787C0c7FC53d",

  "l2GenesisBlockGasLimit": "0x1c9c380",
  "l2GenesisBlockBaseFeePerGas": "0x3b9aca00",
  "l2GenesisRegolithTimeOffset": "0x0",

  "eip1559Denominator": 50,
  "eip1559DenominatorCanyon": 250,
  "eip1559Elasticity": 6,

  "l2GenesisDeltaTimeOffset": null,
  "l2GenesisCanyonTimeOffset": "0x0",

  "systemConfigStartBlock": 0,

  "requiredProtocolVersion": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "recommendedProtocolVersion": "0x0000000000000000000000000000000000000000000000000000000000000000",

  "faultGameAbsolutePrestate": "0x03c7ae758795765c6664a5d39bf63841c71ff191e9189522bad8ebff5d4eca98",
  "faultGameMaxDepth": 44,
  "faultGameMaxDuration": 1200,
  "faultGameGenesisBlock": 0,
  "faultGameGenesisOutputRoot": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "faultGameSplitDepth": 14,

  "preimageOracleMinProposalSize": 1800000,
  "preimageOracleChallengePeriod": 86400,
  "preimageOracleCancunActivationTimestamp": 0
}

**Reply by yohanelly95** [reply]
I have changed the default chainID from 42069 to 12202

**Reply by yohanelly95** [reply]
So can I use OP-Sepolia as the L1 in this case?

**Reply by 1195597989** [reply]
The chain ID 12202 you’ve set for L2 should be correct.

**Reply by yohanelly95** [reply]
I think the optimism branch needs to be updated. It seems https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/scripts/getting-started/config.sh this script is not updated yet with the new variables. @1195597989

**Reply by yohanelly95** [reply]
@k0de26 could this be the issue ?

**Comment by yohanelly95** [comment]
I have copied the `deploy-config/sepolia.json` into my `getting-started.json` and it works now (few variables as mentioned above are missing). Now when I run the command `forge script scripts/Deploy.s.sol:Deploy --sig 'sync()' --rpc-url $L1_RPC_URL` it fails with error. Are all the branches updated, or the OP docs require some updates. 
```
[⠔] Compiling...
No files changed, compilation skipped
Error:
Function `sync()` is not implemented in your script.
```

**Reply by AlfredYyong** [reply]
in `celestia-develop` or `develop` branch, `Deploy.s.sol` has some modifications, so it may not need to executed these command.
I think some of the operations on the OP docs don't quite apply to the `develop` version.

such as in **develop** branch, this func has been removed:
```
function setUp() public virtual override {
        super.setUp();

        string memory path = string.concat(vm.projectRoot(), "/deploy-config/", deploymentContext, ".json");
        cfg = new DeployConfig(path);

        console.log("Deploying from %s", deployScript);
        console.log("Deployment context: %s", deploymentContext);
    }
```

**Reply by yohanelly95** [reply]
Did this command work for you ? @420516460  from here https://docs.optimism.io/builders/chain-operators/tutorials/create-l2-rollup#generate-the-l2-config-files
```
go run cmd/main.go genesis l2 \
  --deploy-config ../packages/contracts-bedrock/deploy-config/getting-started.json \
  --deployment-dir ../packages/contracts-bedrock/deployments/getting-started/ \
  --outfile.l2 genesis.json \
  --outfile.rollup rollup.json \
  --l1-rpc $L1_RPC_URL
  
 ```

**Reply by AlfredYyong** [reply]
no work, i replicated what you did, on the `develop` branch, based on `packages/contracts-bedrock/deploy-config
/sepolia.json` to `getting-started.json`, the contract can be successfully deployed to sepolia, but executing 
`forge script scripts/Deploy.s.sol:Deploy --sig 'sync()' --rpc-url $L1_ RPC_URL`  will show:
```
[⠰] Compiling...
No files changed, compilation skipped
Error: 
Function `sync()` is not implemented in your script.
```
and then i run:
```
go run cmd/main.go genesis l2 \
  --deploy-config ../packages/contracts-bedrock/deploy-config/getting-started.json \
  --deployment-dir ../packages/contracts-bedrock/deployments/getting-started/ \
  --outfile.l2 genesis.json \
  --outfile.rollup rollup.json \
  --l1-rpc $L1_RPC_URL
```
will show:
```
Incorrect Usage: flag provided but not defined: -deployment-dir
...
t=2024-03-26T11:21:13+0000 lvl=crit msg="Application failed" message="flag provided but not defined: -deployment-dir"
exit status 1
```
i m sure you've had similar problems

**Reply by yohanelly95** [reply]
yes faced the same issue

**Comment by yohanelly95** [comment]
This seems to be related to branch is not updated to follow along with the docs, for eg, https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/scripts/getting-started/config.sh does not have the vars fundDevAccounts, faultGameWithdrawalDelay etc which is required by the steps on the Docs.

**Reply by yohanelly95** [reply]
The `tutorials/chain` branch also seems to not have the changes at the moment https://github.com/ethereum-optimism/optimism/blob/tutorials/chain/packages/contracts-bedrock/scripts/getting-started/config.sh

**Reply by yohanelly95** [reply]
cc @sbvegan

**Reply by AlfredYyong** [reply]
exactly agree, actually some changes are not sync to `tutorials/chain`, e,g. https://docs.optimism.io/builders/chain-operators/management/configuration

**Reply by AlfredYyong** [reply]
hi @yohanelly95 , did you success post data to celestia via op-stack?

**Reply by yohanelly95** [reply]
hey @420516460 no I have not done so yet. Currently unable to run the OP Stack Chain node itself. Did you have any success ?

**Reply by AlfredYyong** [reply]
not, i can use the `tutorials/chain` branch to deploy my own L2 and set up the op-batcher to submit to sepolia with `calldata` or `blob` is working. 
but using the `develop` or `celestia-develop` branches has not work yet.

**Reply by sbvegan** [reply]
You're right, the point of the `tutorials/chain` branch is so we didn't have to update the tutorial to be consistent with the latest on `develop` in the monorepo. We just don't have the bandwidth at the moment for that to feasible. The tutorial is supposed to be illustrative and not used for production environments. So we won't be able to keep it up to date for third parties like Celestia. However; with all that being said, please log an issue on our docs repo so we can start figuring out how to prioritizing these updates: https://github.com/ethereum-optimism/docs/issues

**Reply by yohanelly95** [reply]
thanks @sbvegan, could you point me to the changes required in the docs/steps for running an OP Stack chain node ?(changes updated on tutorials/chain and not on develop yet) I can test them, and create a issue + PR to the OP docs.

**Comment by yohanelly95** [comment]
Resolved manually by updating required flags and parameters

**Reply by hardXcoder** [reply]
What was the fix ? Can you share it, I am having the same issue ?
